vllm/model_executor/parallel_utils/pynccl.py

```diff
@@ -40,12 +41,17 @@
         f"Loading nccl from environment variable VLLM_NCCL_SO_PATH={so_file}")
 else:
     if torch.version.cuda is not None:
-        so_file = "libnccl.so.2"
+        cuda_major = torch.version.cuda.split(".")[0]
+        # check if we have vllm-managed nccl
+        path = os.path.expanduser(
+            f"~/.config/vllm/nccl/cu{cuda_major}/libnccl.so.*")
```

Comment:
nit: Let's define this path as a global constant variable?

Reply:
fixed in [21bffe9](https://github.com/vllm-project/vllm/pull/3805/commits/21bffe9b4540088910efb9ab817dbda70ef90a1e)